Measuring Episerver site performance

dratini opened the pull request:
Better than Force Awakens!

Rinorragi commented:
Good post! I did read it but didn't notice anything I would like to change.

Rinorragi commented:
Oh, Use DOTNET as a tag instead of .NET (as others have before you).

ilmoraunio commented:
> but how well do we know the actual performance

of your server?
***
> 10 page request

Probably in plural: requests?
***
> For example, machine with 4 CPU cores should be able to handle more than 144,000 page requests in an hour.

a machine
***
> It is a ready-to-use JMeter project for simple performance testing with aim to minimize manual work

...with _the_ aim...
***
> the only required configuration being test url list

Perhaps: "the only required configurable part being the test url list"?
***
> You can download the GitHub repository as a [ZIP file](https://github.com/dratini/jmeter-perfotrator/archive/master.zip) for convenience, and it should work right out-of-the box, though Java is required, sorry for that.

Ending the sentence at some point would probably give better structure.

Two variations I came up with:

> You can download the GitHub repository as a [ZIP file](https://github.com/dratini/jmeter-perfotrator/archive/master.zip) for convenience, and it should work right out-of-the box. Java is required though, sorry for that.

> You can download the GitHub repository as a [ZIP file](https://github.com/dratini/jmeter-perfotrator/archive/master.zip) for convenience. It should work right out-of-the box, although Java is required. Sorry for that.

***

> Add them to file [_project\urls.txt_](https://github.com/dratini/jmeter-perfotrator/blob/master/project/urls.txt), one per line.

Perhaps "Add the urls to file $url, one url per line." ?
***
> Then fire up JMeter with _start-gui.bat_, you should now see the JMeter UI.

"Fire up JMeter with _start-gui.bat_ and you should see the JMeter UI."
***
> It might look like a Soviet nuclear plant control panel

Somewhat more idiomatic "It might look like a control panel out of a Soviet power plant".
***
> just ensure that Error rate is always zero.

Error % (Couldn't find Error rate)
***
> as it's unit is requests per second

its
***
> Please note that only the html is downloaded; stylesheets, scripts and other embedded resources are ignored.

Needs some context beforehand. You should probably mention that by doing $something1 you'll download $something2 (I am not 100% sure what).
***
> While executing the test, the server CPU usage should be sky-high.

Suggestion: "When you start executing the test"
***
> **View Results in Tree**:Very

Insert space after colon.
***
> Contains http request and response details, thus making this very useful while debugging site access and errors.

It contains

a very useful
***
> but profiling will tell you where is the hold-up.

where the hold-up is.
***
> The code in our hands we can look for fixes

With the code
***
> Like [CachingUrlResolver](https://github.com/solita/episerver-performance/blob/master/Solita.Episerver.Performance/Routing/CachingUrlResolver.cs) that yields massive performance gains with pages containing lots of url resolving.

The point of this sentence is clear (yields perf gains), but I'm having trouble understanding what Like means here? Is it trying to refer to some text or some concept beforehand or is it just glue? I am not sure.

jukkahyv commented on _posts/2016-01-21-measuring-episerver-site-performance.md:
```diff
@@ -0,0 +1,83 @@
+* **View Results in Tree**:Very detailed results per request. Contains http request and response details, thus making this very useful while debugging site access and errors.
+
+### Reliable site setup
+The most reliable results are acquired from an environment identical to production, unfortunately this is not always possible in development stage. 
```
I'd add full stop after "production", so "The most reliable results are acquired from an environment identical to production. Unfortunately this is not always possible in development stage." makes it more natural to read. Or maybe add "but" there.

jukkahyv commented on _posts/2016-01-21-measuring-episerver-site-performance.md:
```diff
@@ -0,0 +1,83 @@
+### Protip: Profiling the code
+JMeter may show you the performance rating, but profiling will tell you where is the hold-up. 
+Very useful for spotting performance issues in code, and to help avoiding premature optimization - just optimize what needs to be optimized. 
+JetBrain's [dotTrace](https://www.jetbrains.com/profiler/) is my tool of choice.
```
It's "JetBrains", so JetBrains'. Or just remove the ' completely, that would work too

jukkahyv commented on _posts/2016-01-21-measuring-episerver-site-performance.md:
```diff
@@ -0,0 +1,83 @@
+JetBrain's [dotTrace](https://www.jetbrains.com/profiler/) is my tool of choice.
+
+### Protip: Acquiring the code
+Sometimes the interesting parts are in external dlls, which may not have source available at all. 
```
Small thing, but maybe "source code" instead of just source

jukkahyv commented on _posts/2016-01-21-measuring-episerver-site-performance.md:
```diff
@@ -0,0 +1,83 @@
+### Protip: Acquiring the code
+Sometimes the interesting parts are in external dlls, which may not have source available at all. 
+Luckily, we have decompilers to solve the problem, for example, JetBrain's [dotPeek](https://www.jetbrains.com/decompiler/). 
+The code in our hands we can look for fixes, workarounds and propose improvements. 
```
"With the code in our hands..."

jukkahyv commented on _posts/2016-01-21-measuring-episerver-site-performance.md:
```diff
@@ -0,0 +1,83 @@
+Luckily, we have decompilers to solve the problem, for example, JetBrain's [dotPeek](https://www.jetbrains.com/decompiler/). 
+The code in our hands we can look for fixes, workarounds and propose improvements. 
+Like [CachingUrlResolver](https://github.com/solita/episerver-performance/blob/master/Solita.Episerver.Performance/Routing/CachingUrlResolver.cs) 
+that yields massive performance gains with pages containing lots of url resolving.
```
URL in allcaps

jukkahyv commented:
It's a good post, I had testing JMeter on my TODO list as well, so that's for this. Added some suggestions.

dratini commented:
Thank you all for comments! I hopefully made all the fixes, or changed the structure to comply :)

Major changes:
- Added a chapter about static resources "Testing static resources"
- Removed the reference to CachingUrlResolver - it was a teaser for upcoming post, but falls a bit too short to be insightful enough on its own.
